dallinger/recruiters.py

```diff
@@ -681,22 +680,37 @@ def reward_bonus(self, assignment_id, amount, reason):
 
     @property
     def is_in_progress(self):
-        # Has this recruiter resulted in any participants?
-        return bool(Participant.query.filter_by(recruiter_id=self.nickname).first())
+        """Does an MTurk HIT for the current experiment ID already exist?"""
+        experiment_id = self.config.get("id")
+        hits = self.mturkservice.get_hits(
+            hit_filter=lambda h: h["annotation"] == experiment_id
+        )
+        try:
+            next(hits)
+        except StopIteration:
+            return False
+        else:
+            return True
 
     @property
     def qualification_active(self):
         return bool(self.config.get("assign_qualifications"))
 
     def current_hit_id(self):
-        any_participant_record = (
-            Participant.query.with_entities(Participant.hit_id)
-            .filter_by(recruiter_id=self.nickname)
-            .first()
+        """Return the ID of the most recent HIT with our experiment ID
+        in the annotation, if any such HITs exist.
+        """
+        experiment_id = self.config.get("id")
+        hits = list(
+            self.mturkservice.get_hits(
+                hit_filter=lambda h: h["annotation"] == experiment_id
+            )
         )
+        if not hits:
+            return None
 
-        if any_participant_record is not None:
-            return str(any_participant_record.hit_id)
+        most_recent = sorted(hits, key=lambda k: k["created"])[-1]
```

Comment:
Should this emit some kind of warning if there are multiple open HITs with the experiment id?

Reply:
I think I was getting ahead of myself a little bit and anticipating potential support for multiple HITs. At this point I think it should warn. I contemplated whether it should raise an error, but I think it's probably better for it to be tolerant, since that scenario (you've somehow got more than one HIT for this exp. ID) might be OK?